slack-lint/src/main/java/slack/lint/denylistedapis/DenyListedApiDetector.kt

```diff
@@ -0,0 +1,459 @@
+      ),
+      DenyListedEntry(
+        className = "androidx.core.content.ContextCompat",
+        functionName = "getDrawable",
+        parameters = listOf("android.content.Context", "int"),
+        errorMessage = "Use Context#getDrawableCompat() instead"
+      ),
+      DenyListedEntry(
+        className = "androidx.core.content.res.ResourcesCompat",
+        functionName = "getDrawable",
+        parameters = listOf("android.content.Context", "int"),
+        errorMessage = "Use Context#getDrawableCompat() instead"
+      ),
+      DenyListedEntry(
+        className = "android.support.test.espresso.matcher.ViewMatchers",
+        functionName = "withId",
+        parameters = listOf("int"),
+        errorMessage =
+        "Consider matching the content description instead. IDs are " +
+          "implementation details of how a screen is built, not how it works. You can't" +
+          " tell a user to click on the button with ID 428194727 so our tests should not" +
+          " be doing that. "
+      ),
+      DenyListedEntry(
+        className = "android.view.View",
+        functionName = "setOnClickListener",
+        parameters = listOf("android.view.View.OnClickListener"),
+        arguments = listOf("null"),
+        errorMessage =
+        "This fails to also set View#isClickable. Use View#clearOnClickListener() instead"
+      ),
+      DenyListedEntry(
+        // If you are deny listing an extension method you need to ascertain the fully qualified
+        // name
+        // of the class the extension method ends up on.
+        className = "kotlinx.coroutines.flow.FlowKt__CollectKt",
+        functionName = "launchIn",
+        errorMessage =
+        "Use the structured concurrent CoroutineScope#launch and Flow#collect " +
+          "APIs instead of reactive Flow#onEach and Flow#launchIn. Suspend calls like Flow#collect " +
+          "can be refactored into standalone suspend funs and mixed in with regular control flow " +
+          "in a suspend context, but calls that invoke CoroutineScope#launch and Flow#collect at " +
+          "the same time hide the suspend context, encouraging the developer to continue working in " +
+          "the reactive domain."
+      ),
+      DenyListedEntry(
+        className = "kotlinx.coroutines.test.TestBuildersKt",
+        functionName = "runBlockingTest",
+        errorMessage =
+        "Use runBlocking or runBlockingWithTestDispatcher instead. runBlockingTest " +
+          "is often overkill. The only difference between it and runBlocking is that it gives you " +
+          "a TestDispatcher. Treat it like RxJava's TestScheduler: if you don't need it, leave it " +
+          "out. But even if you do need it, you are better off using TestDispatcher directly: if a " +
+          "coroutine is blocked on something other than TestDispatcher (which isn't uncommon - " +
+          "most things that block in Cash App aren't coroutines right now), at the end of your " +
```

Comment:
Thoughts about removing the part between () since it might not be relevant to us?

Reply:
Yup good catch, I meant to remove this

Reply:
Just removed this entirely since we don't use it anymore anyway (it's deprecated). 7e991ce